src/ethernet/RsRtspClient.cpp

```diff
@@ -0,0 +1,613 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2017 Intel Corporation. All Rights Reserved.
+
+#include "RsRtspClient.h"
+#include <ipDeviceCommon/RsCommon.h>
+
+#include "liveMedia.hh"
+#include "BasicUsageEnvironment.hh"
+
+#include <iostream>
+#include <thread>
+//#include <condition_variable>
+#include <vector>
+#include <string>
+#include <math.h>
+#include <algorithm>
+
+#define RTSP_CLIENT_VERBOSITY_LEVEL 0 // by default, print verbose output from each "RTSPClient"
+#define REQUEST_STREAMING_OVER_TCP 0  // TODO - uderstand this
+
+std::string format_error_msg(std::string function, RsRtspReturnValue retVal)
+{
+  return std::string("[" + function + "] error: " + retVal.msg + " - " + std::to_string(retVal.exit_code));
+}
+
+long long int RsRTSPClient::getStreamProfileUniqueKey(rs2_video_stream t_profile)
+{
+  long long int key;
+  key = t_profile.type * pow(10, 12) + t_profile.fmt * pow(10, 10) + t_profile.fps * pow(10, 8) + t_profile.index + t_profile.width * pow(10, 4) + t_profile.height;
+  return key;
+}
+
+//TODO: change char* to std::string
+IRsRtsp *RsRTSPClient::getRtspClient(char const *t_rtspURL,
+                                     char const *t_applicationName, portNumBits t_tunnelOverHTTPPortNum)
+{
+  TaskScheduler *scheduler = BasicTaskScheduler::createNew();
+  UsageEnvironment *env = BasicUsageEnvironment::createNew(*scheduler);
+  
+  RTSPClient::responseBufferSize = 100000;
```

Comment:
Add comment. Also - this must be replaced with const and moved to the class definitions/header to maintenance.